repos/spack_repo/builtin/packages/mpibind/package.py

```diff
@@ -46,24 +63,50 @@ class Mpibind(AutotoolsPackage):
     depends_on("pkgconfig", type="build")
 
     depends_on("hwloc@2:+libxml2", type="link")
-    depends_on("hwloc@2:+cuda+nvml", type="link", when="+cuda")
-    depends_on("hwloc@2.4:+rocm+opencl", type="link", when="+rocm")
-    depends_on("hwloc@2:+pci", type="link", when=(sys.platform != "darwin"))
+    depends_on("hwloc@2:+pci", type="link",
+               when=(sys.platform != "darwin"))
+
+    depends_on("hwloc@2: +cuda +nvml", type="link",
+               when="+cuda")
+    depends_on("hwloc@2.4: +rocm +opencl", type="link",
+               when="@:0.19 +rocm")
+    depends_on("hwloc@2.4: +rocm", type="link",
+               when="@0.20: +rocm")
+
+    # Need mpibind v0.23+ and hwloc v2.12+ for NV Grace Hopper
+    depends_on("hwloc@2.12: +cuda +nvml", type="link",
+               when="@0.23: +cuda target=neoverse_v2")
+    conflicts("@:0.22 +cuda target=neoverse_v2",
```

Comment:
Would also need mpibind and hwloc at at least these versions for newer ARM processors? Or is it really *just* `neoverse_v2` that requires this?

```suggestion
               when="@0.23: +cuda target=neoverse_v2:")
    conflicts("@:0.22 +cuda target=neoverse_v2:",
```